Surface_mesh_segmentation/include/CGAL/approximate_convex_decomposition.h

```diff
@@ -0,0 +1,242 @@
+ */
+template <class ConcurrencyTag, class TriangleMesh, class DistancesPropertyMap, class NamedParameters>
+double
+concavity_values(const TriangleMesh& mesh,
+                 DistancesPropertyMap distances,
+                 const NamedParameters& np)
+{
+    typedef typename Polygon_mesh_processing::GetVertexPointMap<TriangleMesh, NamedParameters>::const_type Vpm;
+    typedef typename Polygon_mesh_processing::GetGeomTraits<TriangleMesh, NamedParameters>::type Geom_traits;
+
+    Geom_traits geom_traits = boost::choose_param(boost::get_param(np, internal_np::geom_traits), Geom_traits());
+    Vpm vpm = boost::choose_param(boost::get_param(np, internal_np::vertex_point),
+                                  get_const_property_map(boost::vertex_point, mesh));
+
+    CGAL_precondition(is_triangle_mesh(mesh));
+
+    internal::Concavity<TriangleMesh, Vpm, Geom_traits, ConcurrencyTag> algorithm(mesh, vpm, geom_traits);
+    return algorithm.compute(distances);
+}
+
+
+#ifndef DOXYGEN_RUNNING
+template <class ConcurrencyTag, class TriangleMesh, class DistancesPropertyMap>
+double
+concavity_values(const TriangleMesh& mesh,
+                 DistancesPropertyMap distances)
+{
+    return concavity_values<ConcurrencyTag>(mesh, distances, Polygon_mesh_processing::parameters::all_default());
+}
+
+template <class ConcurrencyTag, class TriangleMesh>
+double
+concavity_values(const TriangleMesh& mesh)
+{
+    CGAL::Static_property_map<typename boost::graph_traits<TriangleMesh>::vertex_descriptor, double > distances_pmap(0);
+    
+    return concavity_values<ConcurrencyTag>(mesh, distances_pmap, Polygon_mesh_processing::parameters::all_default());
+}
+#endif
+
+
+/*!
+ * \ingroup PkgSurfaceSegmentation
+ * @brief Function computing an approximate convex decomposition of a triangle mesh.
+ *
+ * This function fills a property map associating a segment-id to each face (in the range [0, `N-1`]) to each face.
+ * `N` is the number of segments computed by the functions (greater are equal to `min_number_of_segments`).
+ * The set of faces in the same segment defines an edge-connected patch with concavity value less are equal to 'concavity_threshold'.
+ *
+ * @pre `is_triangle_mesh(mesh)`
+ * @pre `faces(mesh).size() >= min_number_of_segments`
+ *
+ * @tparam ConcurrencyTag enables sequential versus parallel algorithm (possible values are `Sequential_tag` and `Parallel_tag`)
+ * @tparam TriangleMesh a model of `FaceListGraph`
+ * @tparam FacePropertyMap a `ReadWritePropertyMap` with the `boost::graph_traits<TriangleMesh>::%face_descriptor` key type and an integer as value type
+ * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+ *
+ * @param mesh triangle mesh on which approximate convex decomposition is computed
+ * @param face_ids property map with per face segment ids
+ * @param concavity_threshold maximal concavity value of the set of faces in a segment
```

Comment:
Some insights must be given to the user: range of values and the effect on the result.